File: GPTTest/gptTest.py
import openai
import config
import sys
# Replace YOUR_API_KEY with your OpenAI API key
openai.api_key = config.CHAT_GPT_PERSONAL_KEY
#Our group key is out of funding or something, so I had to use my own.

userQuery1 = "bubble sort"
userQuery2 = "quick sort"

if(len(sys.argv)==3):
    userQuery1 = sys.argv[1]
    userQuery2 = sys.argv[2]
    

response = openai.ChatCompletion.create(
    model="gpt-3.5-turbo",
    #model = "text-ada:001",
    messages=[
            {"role": "system", "content": "You are a teacher"},
            {"role": "user", "content": "What is the connection between %s and %s?" % (userQuery1, userQuery2)},
        ]
)

result = ''
for choice in response.choices:
    result += choice.message.content

print(result)

File: GPTTest/gptTest.ts
const { Configuration, OpenAIApi } = require("openai");
const readlineSync = require("readline-sync");
require("dotenv").config();

(async () => {
  const configuration = new Configuration({
    apiKey: process.env.OPENAI_API_KEY,
  });
  const openai = new OpenAIApi(configuration);
  const history = [];

  while (true) {
    const user_input1 = readlineSync.question("Your first input: ");
    const user_input2 = readlineSync.question("Your second input: ");
    
    const messages = [];
    for (const [input_text, completion_text] of history) {
      messages.push({ role: "user", content: input_text });
      messages.push({ role: "assistant", content: completion_text });
    }
    var requestString = new String("Tell me the relation between ")
    var fs = requestString.concat(user_input1, " and ", user_input2, ". Keep the response to two paragraphs and start it with yes or no")
    messages.push({ role: "user", content: fs });

    try {
      const completion = await openai.createChatCompletion({
        model: "gpt-3.5-turbo",
        messages: messages,
      });

      const completion_text = completion.data.choices[0].message.content;
      console.log(completion_text);

      history.push([user_input1, completion_text]);

      const user_input_again = readlineSync.question(
        "\nWould you like to continue the conversation? (Y/N)"
      );
      if (user_input_again.toUpperCase() === "N") {
        return;
      } else if (user_input_again.toUpperCase() !== "Y") {
        console.log("Invalid input. Please enter 'Y' or 'N'.");
        return;
      }
    } catch (error) {
      if (error.response) {
        console.log(error.response.status);
        console.log(error.response.data);
      } else {
        console.log(error.message);
      }
    }
  }
})();